src/renderer/camera_controller.cpp

```diff
@@ -1,42 +1,23 @@
 #include "hybrid/renderer/camera_controller.hpp"
-#include "hybrid/hybrid_shared.hpp"
+
 #include "hybrid/core/application.hpp"
+#include "hybrid/hybrid_shared.hpp"
 
 namespace hybrid {
 
-static const glm::vec3 WORLD_UP_VECTOR(0, 1, 0);
-static const glm::vec3 WORLD_FORWARD_VECTOR(0, 0, -1);
-static const glm::vec3 WORLD_RIGHT_VECTOR(-1, 0, 0);
-
-CameraController::CameraController(Camera& camera) : m_camera(camera), m_yaw(0), m_pitch(0), m_position(0,0,3) {}
+CameraController::CameraController(Camera& camera, ModelController& modelController)
```

Comment:
In the future, we may consider to decouple the controllers. Also, we may consider taking the "game development" part to the client side as we discussed. To do those, we can add Unity-like callbacks (start, update) so that the client can make custom development. Just as a note.